chore: increase coverage for uuid

cmwylie19 opened the pull request:
## Description

Increases coverage in the CLI for the UUID command

```bash
 RERUN  src/cli/uuid.test.ts x3 
        Filename pattern: src/cli

 ✓ src/cli/uuid.test.ts (6 tests) 4ms
   ✓ buildUUIDTable > should build table with UUID and description 1ms
   ✓ buildUUIDTable > should include UUID with empty description if none is present 0ms
   ✓ buildUUIDTable > should skip deployments without UUID 0ms
   ✓ getPeprDeploymentsByUUID > should fetch deployments without uuid 0ms
   ✓ getPeprDeploymentsByUUID > should fetch deployments with a specific uuid 0ms
   ✓ uuid CLI command > should log UUID and description when deployment has a UUID 2ms

 Test Files  1 passed (1)
      Tests  6 passed (6)
   Start at  15:39:51
   Duration  270ms

 % Coverage report from v8
----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
----------|---------|----------|---------|---------|-------------------
All files |     100 |      100 |     100 |     100 |                   
 uuid.ts  |     100 |      100 |     100 |     100 |                   
----------|---------|----------|---------|---------|-------------------
```

## Related Issue

Fixes #2403 
<!-- or -->
Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging
- [ ] Unit, [Journey](https://github.com/defenseunicorns/pepr/tree/main/journey), [E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples), [docs](https://github.com/defenseunicorns/pepr/tree/main/docs), [adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or updated as needed
- [ ] [Contributor Guide Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request) followed

cmwylie19 committed:
chore: increase coverage for uuid

Signed-off-by: Case Wylie <cmwylie19@defenseunicorns.com>

cmwylie19 committed:
Merge branch 'main' into 2403

cmwylie19 committed:
Merge branch 'main' into 2403

cmwylie19 committed:
Merge branch 'main' into 2403

cmwylie19 committed:
Merge branch 'main' into 2403

samayer12 commented on src/cli/uuid.test.ts:
```diff
@@ -0,0 +1,132 @@
+          annotations: { "pepr.dev/description": "Test annotation" },
+        },
+      },
+    ],
```
I think these unit tests could benefit from a case where there are multiple deployments, so here's an example of an additional one to add.

```suggestion
      {
        metadata: {
          name: "other-deployment",
          labels: { "pepr.dev/uuid": "asdf" },
          annotations: { "pepr.dev/description": "Another annotation" },
        },
      }
    ],
```

samayer12 commented on src/cli/uuid.test.ts:
```diff
@@ -0,0 +1,132 @@
+  });
+
+  it("should log UUID and description when deployment has a UUID", async () => {
+    await program.parseAsync(["node", "test", "uuid"]);
```
```suggestion
    await program.parseAsync(["uuid"], { from: "user" });
```

I believe this use of `parseAsync` is more clear, since it only focuses on input we'd expect from a user and does not require `node` or `test` to precede our command.

samayer12 commented on src/cli/uuid.test.ts:
```diff
@@ -0,0 +1,132 @@
+    expect(logSpy).toHaveBeenCalledWith("--------------------------------------------");
+    expect(logSpy).toHaveBeenCalledWith("1234\tTest annotation");
+  });
+});
```
```suggestion

  it("should log UUID and description when deployment has a matching UUID", async () => {
    await program.parseAsync(["uuid", "asdf"], { from: "user" });

    expect(logSpy).toHaveBeenCalledWith("UUID\t\tDescription");
    expect(logSpy).toHaveBeenCalledWith("--------------------------------------------");
    expect(logSpy).toHaveBeenCalledWith("asdf\tAnother annotation");
    expect(logSpy).not.toHaveBeenCalledWith("1234\tTest annotation");
  });
});
```

If I understand the behavior of the `uuid` command, I _think_ this should pass. However, it seems that this test case where a user only wants a known module UUID includes all results from the call to `mockK8s.Get()`.

samayer12 reviewed (changes requested):
The tests look good, there's one test case I think we need to consider. Would like your thoughts before approving.

cmwylie19 committed:
Update src/cli/uuid.test.ts

Co-authored-by: Sam Mayer <sam.mayer@defenseunicorns.com>

cmwylie19 committed:
Update src/cli/uuid.test.ts

Co-authored-by: Sam Mayer <sam.mayer@defenseunicorns.com>

cmwylie19 committed:
Merge branch 'main' into 2403

cmwylie19 committed:
Update src/cli/uuid.test.ts

Co-authored-by: Sam Mayer <sam.mayer@defenseunicorns.com>

cmwylie19 committed:
chore: fix broken test and format

Signed-off-by: Case Wylie <cmwylie19@defenseunicorns.com>

cmwylie19 committed:
Merge branch 'main' into 2403

cmwylie19 commented on src/cli/uuid.test.ts:
```diff
@@ -0,0 +1,148 @@
+
+  it("should log UUID and description when deployment has a matching UUID", async () => {
+    await program.parseAsync(["uuid", "asdf"], { from: "user" });
+    console.log(logSpy.mock.calls);
```
```suggestion
```

cmwylie19 committed:
Update src/cli/uuid.test.ts

samayer12 commented on src/cli/uuid.test.ts:
```diff
@@ -0,0 +1,147 @@
+    await program.parseAsync(["uuid", "asdf"], { from: "user" });
+    expect(logSpy).toHaveBeenCalledWith("UUID\t\tDescription");
+    expect(logSpy).toHaveBeenCalledWith("--------------------------------------------");
+    expect(logSpy).toHaveBeenCalledWith("1234\tTest annotation");
```
I'd expect 1234 to not appear in the output when the uuid command is called with `asdf` as the input. Does this track with the intended behavior of the UUID command?

> Module UUID(s) currently deployed in the cluster with their descriptions. [uuid] represents a specific module uuid in the cluster.

cmwylie19 replied:
actually yes you are 100 correct. I will fix that

samayer12 reviewed (changes requested):
One question surrounding a test case where we pass `uuid asdf` as the command. Otherwise, this looks good.

cmwylie19 committed:
chore: PR comments

Signed-off-by: Case Wylie <cmwylie19@defenseunicorns.com>

cmwylie19 committed:
Merge branch 'main' into 2403

cmwylie19 committed:
Merge branch 'main' into 2403

cmwylie19 committed:
Merge branch 'main' into 2403

samayer12 commented on src/cli/uuid.test.ts:
```diff
@@ -0,0 +1,146 @@
+
+    expect(logSpy).toHaveBeenCalledWith("UUID\t\tDescription");
+    expect(logSpy).toHaveBeenCalledWith("--------------------------------------------");
+    expect(logSpy).toHaveBeenCalledWith("1234\tTest annotation");
```
```suggestion
    expect(logSpy).toHaveBeenCalledWith("1234\tTest annotation");
    expect(logSpy).toHaveBeenCalledWith("asdf\tAnother annotation");
```

Our mock contains multiple deployments, so this test case without and filters should show both deployments.

samayer12 commented on src/cli/uuid.test.ts:
```diff
@@ -0,0 +1,146 @@
+    await program.parseAsync(["uuid", "asdf"], { from: "user" });
+    expect(logSpy).toHaveBeenCalledWith("UUID\t\tDescription");
+    expect(logSpy).toHaveBeenCalledWith("--------------------------------------------");
+    expect(logSpy).toHaveBeenCalledWith("asdf\tAnother annotation");
```
```suggestion
    expect(logSpy).not.toHaveBeenCalledWith("1234\tTest annotation");
    expect(logSpy).toHaveBeenCalledWith("asdf\tAnother annotation");
```

This test case focuses on providing a UUID to filter a collection of deployments. So, given a mock that produces two deployments, we should be able to filter it down to just one (`asdf`) to make sure that providing a UUID filters as expected.

Note that current state of the PR causes the test to fail. So, we might need to revisit the implementation details of `getPeprDeploymentsByUUID()`

cmwylie19 replied:
So you are right, and in an integration test that would work but here it is not going to work based on how the mock is set up

```bash
logSpy calls:
[
  [
    "UUID\t\tDescription"
  ],
  [
    "--------------------------------------------"
  ],
  [
    "1234\tTest annotation"
  ],
  [
    "asdf\tAnother annotation"
  ]
]
```

The deployment is always returning two items

```typescript
  const Get = vi.fn().mockResolvedValue({
    items: [
      {
        metadata: {
          name: "test-deployment",
          labels: { "pepr.dev/uuid": "1234" },
          annotations: { "pepr.dev/description": "Test annotation" },
        },
      },
      {
        metadata: {
          name: "other-deployment",
          labels: { "pepr.dev/uuid": "asdf" },
          annotations: { "pepr.dev/description": "Another annotation" },
        },
      },
    ],
  });
  ```

cmwylie19 replied:
ah i see yeah you had just mentioned what I just said, disregard

samayer12 reviewed (changes requested):
It looks like the test case that tests `pepr uuid asdf` is also including the `1234` deployment, which is unexpected.

cmwylie19 committed:
Update src/cli/uuid.test.ts

Co-authored-by: Sam Mayer <sam.mayer@defenseunicorns.com>

cmwylie19 committed:
Merge branch 'main' into 2403

samayer12 committed:
Merge branch 'main' into 2403

samayer12 committed:
chore: test updates for PR 2403 (#2458)

## Description

This PR updates the test conditions used in #2403 

## Related Issue

Relates to #2403 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging
- [x] Unit,
[Journey](https://github.com/defenseunicorns/pepr/tree/main/journey),
[E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples),
[docs](https://github.com/defenseunicorns/pepr/tree/main/docs),
[adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or
updated as needed
- [x] [Contributor Guide
Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request)
followed

samayer12 committed:
Merge branch 'main' into 2403
